### PR TITLE
Upgrade py-awscli to 1.11.28

### PR DESCRIPTION
**Pull request by milkcocoa**
I've updated awscli from 1.9.21 to 1.11.24, and dependent stuffs. s3transfer is newly added.

### CODE REVIEW

**Comment by larryv** [comment]
This commit should be broken up because it affects many ports in unrelated ways. I will do this for you.

**Comment by larryv** [comment]
I committed the changes to nomaintainer ports and split up the rest.

File: python/py-awscli/Portfile [hunk 1]
```diff
@@ -30,15 +29,12 @@ if {${name} ne ${subport}} {
...
     depends_lib-append \
-                        port:py${python.version}-asn1 \
-                        port:py${python.version}-bcdoc \
                         port:py${python.version}-botocore \
                         port:py${python.version}-colorama \
-                        port:py${python.version}-dateutil \
                         port:py${python.version}-docutils \
-                        port:py${python.version}-jmespath \
                         port:py${python.version}-rsa \
-                        port:py${python.version}-six
```
Comment (larryv) [inline python/py-awscli/Portfile]: Did you remove these dependencies because botocore includes them transitively? Or does AWS CLI actually not require them anymore?

File: python/py-botocore/Portfile [hunk 2]
```diff
@@ -17,17 +17,18 @@ homepage            https://github.com/boto/botocore
...
 if {${name} ne ${subport}} {
-    depends_build-append \
-                        port:py${python.version}-setuptools
+    depends_build-append    port:py${python.version}-setuptools
+    depends_lib-append      port:py${python.version}-dateutil \
+                            port:py${python.version}-docutils \
+                            port:py${python.version}-jmespath
 
     test.run            yes
-#    test.target         check
```
Comment (larryv) [inline python/py-botocore/Portfile]: Why did you comment this out?

File: python/py-s3transfer/Portfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+distname            ${python.rootname}-${version}
+checksums           md5     036d467e77e0d4dfa65448d63a9208bc \
+                    rmd160  cdbd57c8617cf6a3f38a3049540d18b7d42c435a \
+                    sha256  17ad7d672115f93a72ed7917209cb0bb02fc87f96f11886408ed8a6b1bb4c754
+
+python.versions     27 34
+
+if {${name} ne ${subport}} {
+    depends_build-append    port:py${python.version}-setuptools
+    depends_lib-append      port:py${python.version}-futures
```
Comment (larryv) [inline python/py-s3transfer/Portfile]: This is missing dependencies—at least botocore, if not more.

**Comment by pixilla** [comment]
Please remove the deprecated second row "# $Id$" tag.

**Comment by larryv** [comment]
@pixilla I have removed the `$Id$` lines.

File: python/py-awscli/Portfile (hunk 1 above)
Comment (milkcocoa) [inline python/py-awscli/Portfile]: Because botocore includes them transitively, and at `requires.txt` of awscli, it requires only those dependencies.

**Comment by milkcocoa** [comment]
> I committed the changes to nomaintainer ports and split up the rest.

Thank you, @larryv.

**Comment by milkcocoa** [comment]
Did you decide not to update py-yaml?

**Comment by milkcocoa** [comment]
Oops, it seems that I forgot including update for py-yaml, sorry. I'll create another pull request.

File: python/py-awscli/Portfile (hunk 1 above)
Comment (larryv) [inline python/py-awscli/Portfile]: A port **must** declare dependencies on all other software it uses, even if another dependency includes them also. Transitive dependencies are very fragile (and misleading besides).

I did a cursory textual search for a couple of the removed dependencies. Looks like some of the "customizations" scripts use them, but I don't know what those scripts are for or whether they're installed. We might be able to ignore them entirely.

File: python/py-botocore/Portfile (hunk 2 above)
Comment (larryv) [inline python/py-botocore/Portfile]: Oh sorry, I misread the diff. You deleted a line that was _already_ commented out. Got it.

File: python/py-awscli/Portfile (hunk 1 above)
Comment (milkcocoa) [inline python/py-awscli/Portfile]: > A port must declare dependencies on all other software it uses, even if another dependency includes them also. 

I've got it. So could you revert this change?

**Comment by milkcocoa** [comment]
I've reverted deleting dependencies, but I've removed only `py-bcdoc` because `py-bcdoc` is [a document generation tool for botocore](https://pypi.python.org/pypi/bcdoc), but it isn't listed on `requirements.txt` of `botocore`.

**Comment by milkcocoa** [comment]
@larryv @pixilla How is the status of this pull request? Do I need another update?

**Comment by larryv** [comment]
Sorry, I did not do any MacPorts work this past week.

I would like to clear up this dependency business. The following modules in `requirements.txt` are listed as dependencies in the new portfile. This is fine.
- botocore
- colorama
- docutils
- jmespath
- PyYAML (`yaml`)
- rsa
- s3transfer

The following modules in `requirements.txt` are not listed as dependencies. I guess nose, mock, and tox are only needed for testing? What is wheel for?
- nose
- mock
- tox
- wheel

The following dependencies are not listed in `requirements.txt`. If AWS CLI does not use them directly, they can be removed.
- asn1
- dateutil
- six

**Comment by milkcocoa** [comment]
Unfortunately, I'm not familiar with Python package management, but I've tried to install it with pip locally. Here is the log.

```
$ pip install awscli
Collecting awscli
  Downloading awscli-1.11.28-py2.py3-none-any.whl (1.0MB)
    100% |████████████████████████████████| 1.1MB 878kB/s
Collecting docutils>=0.10 (from awscli)
  Downloading docutils-0.13.1-py2-none-any.whl (537kB)
    100% |████████████████████████████████| 542kB 1.4MB/s
Collecting PyYAML<=3.12,>=3.10 (from awscli)
  Downloading PyYAML-3.12.tar.gz (253kB)
    100% |████████████████████████████████| 256kB 2.7MB/s
Collecting colorama<=0.3.7,>=0.2.5 (from awscli)
  Downloading colorama-0.3.7-py2.py3-none-any.whl
Collecting s3transfer<0.2.0,>=0.1.9 (from awscli)
  Downloading s3transfer-0.1.9-py2.py3-none-any.whl (53kB)
    100% |████████████████████████████████| 61kB 5.9MB/s
Collecting botocore==1.4.85 (from awscli)
  Downloading botocore-1.4.85-py2.py3-none-any.whl (3.1MB)
    100% |████████████████████████████████| 3.1MB 364kB/s
Collecting rsa<=3.5.0,>=3.1.2 (from awscli)
  Downloading rsa-3.4.2-py2.py3-none-any.whl (46kB)
    100% |████████████████████████████████| 51kB 7.8MB/s
Collecting futures<4.0.0,>=2.2.0; python_version == "2.6" or python_version == "2.7" (from s3transfer<0.2.0,>=0.1.9->awscli)
  Downloading futures-3.0.5-py2-none-any.whl
Collecting python-dateutil<3.0.0,>=2.1 (from botocore==1.4.85->awscli)
  Downloading python_dateutil-2.6.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 3.1MB/s
Collecting jmespath<1.0.0,>=0.7.1 (from botocore==1.4.85->awscli)
  Downloading jmespath-0.9.0-py2.py3-none-any.whl
Collecting pyasn1>=0.1.3 (from rsa<=3.5.0,>=3.1.2->awscli)
  Downloading pyasn1-0.1.9-py2.py3-none-any.whl
Collecting six>=1.5 (from python-dateutil<3.0.0,>=2.1->botocore==1.4.85->awscli)
  Downloading six-1.10.0-py2.py3-none-any.whl
Building wheels for collected packages: PyYAML
  Running setup.py bdist_wheel for PyYAML ... done
  Stored in directory: /Users/shu/Library/Caches/pip/wheels/2c/f7/79/13f3a12cd723892437c0cfbde1230ab4d82947ff7b3839a4fc
Successfully built PyYAML
Installing collected packages: docutils, PyYAML, colorama, futures, six, python-dateutil, jmespath, botocore, s3transfer, pyasn1, rsa, awscli
Successfully installed PyYAML-3.12 awscli-1.11.28 botocore-1.4.85 colorama-0.3.7 docutils-0.13.1 futures-3.0.5 jmespath-0.9.0 pyasn1-0.1.9 python-dateutil-2.6.0 rsa-3.4.2 s3transfer-0.1.9 six-1.10.0
```

From this log, awscli doesn't use the following packages directly.

* asn1 (from rsa)
* dateutil (from botocore)
* six (from dateutils)

> I guess nose, mock, and tox are only needed for testing?

I think so, too. nose is a unit test framework, and tox is a command line tool for testing.

> What is wheel for?

wheel seems to be a tool which is complement for pip, and is automatically installed with pip.

**Comment by larryv** [comment]
Okay, I'll push an updated version. They're up to 1.11.28 by now 😒

**Comment by larryv** [comment]
Just force-pushed an updated commit series that includes botocore 1.4.85 and awscli 1.11.28. Please test.

**Comment by milkcocoa** [comment]
Thanks. I'll test this version.

**Comment by milkcocoa** [comment]
I've tried to install this version, and succeeded to show version as follows.

```
aws --version
aws-cli/1.11.28 Python/2.7.12 Darwin/16.3.0 botocore/1.4.81
```

botocore is also updated to 1.4.85, should I need to create another pull request?

**Comment by milkcocoa** [comment]
Thank you. I missed the change, and upgraded botocore, too.

```
aws --version
aws-cli/1.11.28 Python/2.7.12 Darwin/16.3.0 botocore/1.4.85
```

This looks good.

**Comment by tobypeterson** [comment]
Not sure what the delay is here, everyone above seems to approve. Going to go ahead and merge....
